Advanced network support

leszko opened the pull request:
The main reason for this PR is to enabled using Advanced Network (and therefore SSL Mutual Authentication with health check on different non-ssl port).

Changes:
- Add `livenessProbe.path`, `livenessProbe.port`, `livenessProbe.scheme` (add the same for `readinessProbe`)
- Change `readinessProbe` endpoint to `/ready`
- Remove `hazelcast.rest` property (this is a non-backward-compatible change, however we need to do it, because 1) Advanced Network cannot work with this parameter 2) It's deprecated and actually removed [here](https://github.com/hazelcast/hazelcast/pull/15981/))
- Make REST disabled by default (non-backwards-compatible change, but makes the cluster more secure by default).

Note that:
- I have tested it works correctly with Advanced Network
- I have tested it does not drop any data while Rolling Upgrade
- It's the continuation of the work of @galibey (#79). Thanks Alexander for starting that. Good work! 👍 
- I applied all the comments from #79

mesutcelik commented on stable/hazelcast-enterprise/README.md:
```diff
@@ -53,7 +53,6 @@ The following table lists the configurable parameters of the Hazelcast chart and
 | `cluster.memberCount`                      | Number of Hazelcast members                                                                                    | 2                                                    |
 | `hazelcast.licenseKey`                     | Hazelcast Enterprise License Key                                                                               | `nil`                                                |
 | `hazelcast.licenseKeySecretName`           | Kubernetes Secret Name, where Hazelcast Enterprise License Key is stored (can be used instead of licenseKey)   | `nil`                                                |
-| `hazelcast.rest`                           | Enable REST endpoints for Hazelcast member                                                                     | `true`                                               |
```
What is the alternative now for users who were using REST API like `cluster.sh` or `healthcheck.sh`? Configuring REST via advanced networking in hazelcast.yaml?

leszko replied:
added `Notable Changes` section to README with the explanation.

leszko commented on stable/hazelcast-enterprise/templates/statefulset.yaml:
```diff
@@ -69,9 +69,9 @@ spec:
-            port: {{ if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}5701{{ end }}
-            scheme: {{ if .Values.hazelcast.ssl }}HTTPS{{ else }}HTTP{{ end }}
+            path: {{ .Values.livenessProbe.path }}
+            port: {{ if .Values.livenessProbe.port }}{{ .Values.livenessProbe.port }}{{ else if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}5701{{ end }}
```
No, `.Values.linvenessProbe.port` is not defined by default. So, unless you need it, it's not defined.

mesutcelik replied:
ah ok. It is commented out in values.yaml. What about uncomment it with default value like `port:5701`
then you condition would be simpler.
`{{ if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}{{.Values.livenessProbe.port  }} {{ end }}`

leszko replied:
It's not the same. If you have both `hostPort` and `livenessProbe.port` defined (that will be actually the case in HZ Cloud), then we should use `livenessProbe.port` for health check.

leszko commented on stable/hazelcast-enterprise/templates/statefulset.yaml:
```diff
@@ -69,9 +69,9 @@ spec:
         {{- if .Values.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /hazelcast/health/node-state
```
right, changed.

leszko commented:
> Hi @leszko , @galibey ,
> I added my comments. Do we have somewhere we list backward incompatibility changes like release notes. It could be needed for existing users.

Added `Notable Changes` to README.
